android/build.gradle

```diff
@@ -31,30 +76,49 @@ android {
     }
 
     defaultConfig {
-        minSdkVersion safeExtGet('minSdkVersion', 16)
-        targetSdkVersion safeExtGet('targetSdkVersion', 28)
-
+        minSdkVersion safeExtGet('minSdkVersion', 21)
+        targetSdkVersion safeExtGet('targetSdkVersion', 34)
         versionCode 2
         versionName "1.1"
+        buildConfigField "boolean", "IS_NEW_ARCHITECTURE_ENABLED", isNewArchitectureEnabled().toString()
     }
     lintOptions {
        warning 'InvalidPackage', 'MissingPermission'
     }
     testOptions {
         unitTests.returnDefaultValues = true
     }
+
+    sourceSets.main {
+        java {
+            if (isNewArchitectureEnabled()) {
+                srcDirs += [
+                    "src/newArch/java",
+                ]
+            } else {
+                srcDirs += [
+                    "src/oldArch/java",
+                ]
+            }
+        }
+    }
 }
 
 repositories {
-    mavenCentral()
-    google()
-    maven {
-        // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
-        url "$rootDir/../node_modules/react-native/android"
-    }
+  maven {
+      // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
+      url "$projectDir/../node_modules/react-native/android"
+  }
+  google()
+  mavenCentral()
 }
 
 dependencies {
+  if (isNewArchitectureEnabled() && getReactNativeMinorVersion() < 71) {
```

Comment:
It doesn't make sense IMO since we never really support more than 2 versions of RN on new arch. I'd leave only `    implementation 'com.facebook.react:react-native:+'`